src/core/lib/surface/call.c

```diff
@@ -984,6 +984,10 @@ static void continue_receiving_slices(grpc_exec_ctx *exec_ctx,
       call->receiving_message = 0;
       grpc_byte_stream_destroy(exec_ctx, call->receiving_stream);
       call->receiving_stream = NULL;
+      if (call->status[STATUS_FROM_API_OVERRIDE].code != GRPC_STATUS_OK) {
```

Comment:
Yeah, I don't like how it smells (having had to choose STATUS_FROM_API_OVERRIDE for lack of something else). Also testing for != OK. How about having a bool in the call instance that's set by `grpc_call_cancel` for purposes of internal cleanups like this one?